### DISCUSSION TITLE
Speed up & disable cargo check

### DISCUSSION THREAD

**Discussion by thewh1teagle** [rust-lang/rust-analyzer#16355] [Q&A]
How can I disable cargo check of rust-analyzer?
Most of the time I don't want rust-analyzer to help me and I stop the server because it's so slow...
but it keeps annoying in `vscode`, everytime I save file it runs `cargo check`

Also, if you have idea how to speed up `rust-analyzer` please let me know, 
I gave up and developing without it.

**Comment by davidbarsky** [comment]
To disable Cargo check functionality, you'll want to set `rust-analyzer.checkOnSave` to `false` (see [the manual for details](https://rust-analyzer.github.io//manual.html).

You might be able to speed things up by setting `rust-analyzer.cachePriming.enable` to `false`, but I'm not sure if the slowness you're experiencing is on autocomplete or rust-analyzer loading your project. Can you share your machine's hardware specifications?

**Reply by thewh1teagle** [reply]
I already configured `checkOnSave` to `false`. the project is open source [here](https://github.com/thewh1teagle/vibe)


PC spec:
Model: `Xiaomi Redmibook 14 2020`
CPU: `AMD Ryzen 5 4500u`
RAM: `16GB 2400MHZ`
Disk Type: `NVME`
